web/js/components/measure-tool/measure-button.js

```diff
@@ -21,110 +20,81 @@ const MEASURE_MENU_PROPS = {
 const mobileHelpMsg = 'Tap to add a point. Double-tap to complete.';
 const helpMsg = 'Click: Add a point. Right-click: Cancel. Double-click to complete. ';
 
-class MeasureButton extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showAlert: true,
-      isTouchDevice: false,
-    };
-    this.onButtonClick = this.onButtonClick.bind(this);
-    this.dismissAlert = this.dismissAlert.bind(this);
-  }
+const MeasureButton = memo(() => {
+  const dispatch = useDispatch();
 
-  onButtonClick(evt) {
-    const { openModal } = this.props;
-    const isTouchDevice = evt.type === 'touchend';
+  const { isActive, isDistractionFreeModeActive, isMobile } = useSelector((state) => ({
+    isActive: state.measure.isActive,
+    isDistractionFreeModeActive: state.ui.isDistractionFreeModeActive,
+    isMobile: state.screenSize.isMobileDevice,
+  }), shallowEqual);
+
+  const [showAlert, setShowAlert] = useState(true);
+  const [isTouchDevice, setIsTouchDevice] = useState(false);
+
+  const dismissAlert = () => setShowAlert(false);
+
+  const onButtonClick = (evt) => {
+    const touchDevice = evt.type === 'touchend';
```

Comment:
I would prefer to see these lines reordered as below. I know this functionally is the same but I find it easier to parse code if we handle one concept at a time. By moving line 38 down 2 lines we address button behavior first & then move on to device type instead of having them interwoven. 

```
const onButtonClick = (evt) => {
    evt.stopPropagation();
    evt.preventDefault();
    const touchDevice = evt.type === 'touchend';
    MEASURE_MENU_PROPS.touchDevice = isTouchDevice;
```